### PR TITLE
docs: rework the i18n documentation for clarity and remove obsolete parts

### PR DESCRIPTION
**Pull request by dachary**
## Status

Ready for review

## Description of Changes

rework the i18n documentation for clarity and remove obsolete parts

## Testing

* cd securedrop ; make docs
* firefox localhost:8000

It may help to read the [i18n code walkthough](https://forum.securedrop.club/t/recuring-securedrop-code-walkthrough/398/3)

## Deployment

N/A

### If you made changes to documentation:

- [x] Doc linting passed locally

### CODE REVIEW

File: docs/development/i18n.rst [hunk 1]
```diff
@@ -2,23 +2,65 @@ Internationalization (i18n)
...
+wrapped with `gettext`_ functions to be substituted with the
+equivalent localized string. For instance:
+
+.. code:: python
+
+     if not (msg or fh):
+       flash(gettext("You must enter a message or choose a file to submit."), "error")
+       return redirect(url_for('main.lookup'))
+
+The ``gettext`` function name is used as
```
Comment (heartsucker) [inline docs/development/i18n.rst]: This sounds like it's *only* used as a marker. Maybe say "extractors look for calls to the function `gettext` when collecting strings for translation."

Comment (dachary) [reply docs/development/i18n.rst]: done

**Comment by dachary** [comment]
Fixed & rebased & repushed. How does it look now ?

**Comment by redshiftzero** [comment]
thanks for writing this @dachary! were your comments addressed @heartsucker?

**Pull request merged**